### PR TITLE
change(devops): Reduce number of dependabot PRs

### LINKED ISSUE #6547: Group related dependency updates to avoid selecting multiple versions of the same transitive dependencies
## Motivation

Dependabot PRs often need to be grouped together manually to avoid adding then promptly removing entries from the deny.toml skip list.

We could automate this so that the bot groups related dependency updates into a single PR on its own.

### Notes

Dependabot doesn't support this yet (see https://github.com/dependabot/dependabot-core/issues/1190), but [Renovate](https://docs.renovatebot.com/rust/) can group related dependencies when explicitly configured.

We could wait for this feature to be added to Dependabot or switch to Renovate.

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We're getting a lot of dependabot PRs, but some of them don't have a meaningful amount of changes. Others need to be grouped to pass CI.

Close #6547

### Specifications

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups

### Complex Code or Requirements

We can update the groups later if they turn out not to work.

## Solution

- Only check for dependency updates weekly on Mondays, rather than daily (some dependencies update multiple times per week)
- Group Rust dependency updates that usually happen together
- Reduce the maximum number of open PRs to `2 * (reviewers - 1)`, because often someone is away or busy

### Testing

- add a validate dependabot step to our existing action lint job

## Review

Let's try this and see how it goes?

### Reviewer Checklist

  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

Gustavo (or someone else) might want to group related GitHub Actions dependency updates, but I'm not sure what those groups should be.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -2,21 +2,141 @@ version: 2
 updates:
   - package-ecosystem: cargo
     directory: '/'
+    # serde, clap, ans other dependencies sometimes have multiple updates in a week
```
Comment (upbqdn) [inline .github/dependabot.yml]:
```suggestion
    # serde, clap, and other dependencies sometimes have multiple updates in a week
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -2,21 +2,141 @@ version: 2
...
+    open-pull-requests-limit: 8
     labels:
       - 'C-trivial'
       - 'A-rust'
       - 'A-dependencies'
       - 'P-Low :snowflake:'
+    groups:
+        ecc:
+          patterns:
+            # deliberately include zcash_script
```
Comment (upbqdn) [inline .github/dependabot.yml]: What do you mean by this comment?

**Review by upbqdn — COMMENTED** [review]
Looks great. Thanks for creating the groups.

**Review by upbqdn — DISMISSED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (teor2345) [reply .github/dependabot.yml]: This might help?

```suggestion
            # deliberately include zcash_script (even though it is maintained by ZF)
```

**Comment by teor2345** [comment]
@mergifyio update

**Review by upbqdn — APPROVED** [review]

**Comment by upbqdn** [comment]
lol 
image

**Comment by teor2345** [comment]
> lol
> image

I think I tried to fix this in PR #7158, some of the auto-label settings are too much 🤣

**Pull request merged**